### PR TITLE
Translate breakpoint locations to original locations

### PR DESCRIPTION
**Pull request by jasonLaster**
I can't believe how much cleaner this approach is from before. 

Things fell in line really nicely:
+ selectors translate locations
+ source map acquired a couple useful helpers
+ client is wrapped so that it receives generated locations

This probably could be dried up further, but it shows promise

### CODE REVIEW

File: public/js/actions/breakpoints.js [hunk 1]
```diff
@@ -31,9 +25,17 @@ function _getOrCreateBreakpoint(state, location, condition) {
   return getBreakpoint(state, location) || fromJS({ location, condition });
 }
 
+function enableBreakpoint(location: Location) {
```
Comment (jasonLaster) [inline public/js/actions/breakpoints.js]: moved below the helper functions

File: public/js/actions/breakpoints.js [hunk 2]
```diff
@@ -31,9 +25,17 @@ function _getOrCreateBreakpoint(state, location, condition) {
...
+  // Enabling is exactly the same as adding. It will use the existing
+  // breakpoint that still stored.
+  return addBreakpoint(location);
+}
+
 function addBreakpoint(location: Location,
                        { condition, getTextForLine } : any = {}) {
   return ({ dispatch, getState, client }: ThunkArgs) => {
+    location = getOriginalLocation(getState(), location);
```
Comment (jlongster) [inline public/js/actions/breakpoints.js]: Do we need to do this here? Any location coming from the UI should already be an original location, but if you want to do it for sanity, that's fine. Once we actually type locations we should be able to enforce that `location` is an original location and we shouldn't need to do this.

File: public/js/selectors.js [hunk 3]
```diff
@@ -99,6 +103,43 @@ function getSourceMapURL(state: AppState, source: Source) {
   return tab.get("url") + "/" + source.sourceMapURL;
 }
 
+function getGeneratedLocation(state: AppState, location: Location) {
```
Comment (jlongster) [inline public/js/selectors.js]: Perfect, this looks great

**Comment by jlongster** [comment]
👍  This looks awesome and does seem to have the feeling I would expect sourcemaps to have. Great work!

File: public/js/actions/breakpoints.js (hunk 2 above)
Comment (jasonLaster) [inline public/js/actions/breakpoints.js]: yup, it was sanity. We can hold off on it for now

**Comment by jasonLaster** [comment]
Thanks. Yeah. I'm really happy with how this looks.

**Pull request merged**